File: app/src/main/java/org/metabrainz/mobile/presentation/features/userdata/UserViewModel.java
package org.metabrainz.mobile.presentation.features.userdata;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import org.metabrainz.mobile.data.sources.api.entities.mbentity.MBEntity;

public class UserViewModel extends ViewModel {

    private final MutableLiveData<MBEntity> entityData = new MutableLiveData<>();

    public UserViewModel() {
    }

    LiveData<MBEntity> getUserData() {
        return entityData;
    }

    public void setUserData(MBEntity entity) {
        entityData.setValue(entity);
    }
}

File: app/src/main/java/org/metabrainz/mobile/presentation/features/userdata/UserViewModel.kt
package org.metabrainz.mobile.presentation.features.userdata

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import org.metabrainz.mobile.data.sources.api.entities.mbentity.MBEntity

class UserViewModel : ViewModel() {

    private val entityData = MutableLiveData<MBEntity>()

    val userData: LiveData<MBEntity> = entityData

    fun setUserData(entity: MBEntity) {
        entityData.value = entity
    }
}